server/src/main/java/org/elasticsearch/index/codec/tsdb/ES97TSDBDocValuesConsumer.java

```diff
@@ -0,0 +1,213 @@
+            meta.writeInt(directMonotonicBlockShift);
+            final ByteBuffersDataOutput indexOut = new ByteBuffersDataOutput();
+            final DirectMonotonicWriter indexWriter = DirectMonotonicWriter.getInstance(
+                meta,
+                new ByteBuffersIndexOutput(indexOut, "temp-dv-index", "temp-dv-index"),
+                1L + ((numValues - 1) >>> numericBlockShift),
+                directMonotonicBlockShift
+            );
+
+            final long[] buffer = new long[numericBlockSize];
+            int bufferSize = 0;
+            final long valuesDataOffset = data.getFilePointer();
+            final ES97TSDBDocValuesEncoder encoder = new ES97TSDBDocValuesEncoder(numericBlockSize);
+
+            values = valuesProducer.getSortedNumeric(field);
+            for (int doc = values.nextDoc(); doc != DocIdSetIterator.NO_MORE_DOCS; doc = values.nextDoc()) {
+                final int count = values.docValueCount();
+                for (int i = 0; i < count; ++i) {
+                    buffer[bufferSize++] = values.nextValue();
+                    if (bufferSize == numericBlockSize) {
+                        indexWriter.add(data.getFilePointer() - valuesDataOffset);
+                        encoder.encode(buffer, data);
+                        bufferSize = 0;
+                    }
+                }
+            }
+            if (bufferSize > 0) {
+                indexWriter.add(data.getFilePointer() - valuesDataOffset);
+                // Fill unused slots in the block with zeroes rather than junk
+                Arrays.fill(buffer, bufferSize, numericBlockSize, 0L);
+                encoder.encode(buffer, data);
+            }
+
+            final long valuesDataLength = data.getFilePointer() - valuesDataOffset;
+            indexWriter.finish();
+            final long indexDataOffset = data.getFilePointer();
+            data.copyBytes(indexOut.toDataInput(), indexOut.size());
+            meta.writeLong(indexDataOffset);
+            meta.writeLong(data.getFilePointer() - indexDataOffset);
+
+            meta.writeLong(valuesDataOffset);
+            meta.writeLong(valuesDataLength);
+        }
+
+        return new long[] { numDocsWithValue, numValues };
+    }
+
+    @Override
+    public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
+        throw new UnsupportedOperationException("Unsupported binary doc values for field [" + field.name + "]");
+    }
+
+    @Override
+    public void addSortedField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
+        throw new UnsupportedOperationException("Unsupported sorted doc values for field [" + field.name + "]");
+    }
+
+    @Override
+    public void addSortedNumericField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
+        throw new UnsupportedOperationException("Unsupported sorted numeric doc values for field [" + field.name + "]");
+    }
```

Comment:
We need to support this one. "SORTED_NUMERIC" just means "multi-valued" to Lucene. The SORTED bit comes from the fact that values must be returned in order.